frontend/src/components/data-products/data-product-form/data-product-form.component.tsx

```diff
@@ -220,14 +249,63 @@ export function DataProductForm({ mode, dataProductId }: Props) {
             >
                 <Input />
             </Form.Item>
-            {/*Disabled field that will render an external ID everytime the name changes*/}
             <Form.Item<DataProductCreateFormSchema>
+                label={t('Namespace')}
+                tooltip={t('The namespace of the data product')}
                 required
-                name={'external_id'}
-                label={t('External ID')}
-                tooltip={t('The external ID of the data product')}
             >
-                <Input disabled />
+                <Space.Compact direction="horizontal" className={styles.namespace}>
+                    <Form.Item
+                        name={'namespace'}
+                        noStyle
+                        hasFeedback
+                        validateFirst
+                        validateDebounce={DEBOUNCE}
+                        rules={[
+                            {
+                                required: canEditNamespace,
+                                message: t('Please input the namespace of the data product'),
+                            },
+                            {
+                                validator: async (_, value) => {
+                                    if (mode === 'edit' || (!canEditNamespace && !value)) {
+                                        return Promise.resolve();
+                                    }
+
+                                    const validationResponse = await validateNamespace(value.toLowerCase()).unwrap();
```

Comment:
This logic is the same everywhere right? Maybe we can split this in a namespace component? Or at least split the validation handling towards an extracted function